modules/netns-chef.nix

```diff
@@ -0,0 +1,257 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.nix-bitcoin.netns-isolation;
+
+  netns = builtins.mapAttrs (n: v: {
+    inherit (v) id;
+    address = "169.254.${toString cfg.addressblock}.${toString v.id}";
+    availableNetns = builtins.filter (x: config.services.${x}.enable) availableNetns.${n};
+  }) enabledServices;
+
+  # Symmetric netns connection matrix
+  # if clightning.connections = [ "bitcoind" ]; then
+  #   availableNetns.bitcoind = [ "clighting" ];
+  #   and
+  #   availableNetns.clighting = [ "bitcoind" ];
+  availableNetns = let
+    # base = { clightning = [ "bitcoind" ]; ... }
+    base = builtins.mapAttrs (n: v:
+      builtins.filter isEnabled v.connections
+    ) enabledServices;
+  in
+    foldl (xs: s1:
+      foldl (xs: s2:
+        xs // { "${s2}" = xs.${s2} ++ [ s1 ]; }
+      ) xs cfg.services.${s1}.connections
+    ) base (builtins.attrNames base);
+
+  enabledServices = filterAttrs (n: v: isEnabled n) cfg.services;
+  isEnabled = x: config.services.${x}.enable;
+
+  ip = "${pkgs.iproute}/bin/ip";
+  iptables = "${pkgs.iptables}/bin/iptables";
```

Comment:
Consider using `config.networking.firewall.package` instead for nftables. (Though if nixos/nixpkgs#81172 should make this uneccessary.)

Reply:
Is nftables better than iptables for what we're trying to do?

Reply:
Sorry, I meant to write "... instead for *compatibility with* nftables". Using `config.networking.firewall.package` instead of a concrete package would make it work regardless of whether the system is set to use iptables or nftables.

IMO nftables is the future but both work fine for this use case.

Reply:
See `fixup! architecture: pkgs.iptables -> config.networking.firewall.package`